.github/workflows/tests.yml

```diff
@@ -0,0 +1,201 @@
+        if: runner.os == 'Linux'
+        run: sudo apt-get -y install valgrind
+
+      - name: Setup Java ☕
+        uses: actions/setup-java@v3.4.1
+        with:
+          distribution: temurin
+          java-version: 8
+          cache: sbt
+
+      - name: Setup Python 🐍
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - name: Install Conan 💪
+        id: conan
+        uses: turtlebrowser/get-conan@main
+
+      - name: Conan version 💪
+        run: echo "${{ steps.conan.outputs.version }}"
+
+      - name: Setup cmake 🔧
+        uses: lukka/get-cmake@latest
+
+      - name: Prepare, Build, and Install Ωedit 🔧
+        run: |
+          cmake -S . -B build -DCMAKE_BUILD_TYPE=Debug -DBUILD_SHARED_LIBS=YES --install-prefix "${PWD}/_install"
+          cmake --build build
+          cmake --install build/packaging --prefix "${PWD}/_install" --config Debug
+
+      - name: Copy library file mac/linux 🚚
+        if: runner.os != 'Windows'
+        run: cp -r _install/lib/* ./lib
+        shell: bash
+
+      - name: Copy library file windows 🚚
+        if: runner.os == 'Windows'
+        run: cp -r _install/bin/omega_edit.dll lib/
+        shell: bash
+
+      - name: Test Ωedit RPC on macOS 📋
+        if: runner.os == 'macOS'
+        run: |
+          build/bin/server_test
+
+      - name: Test Ωedit RPC on Linux 📋
+        if: runner.os == 'Linux'
+        run: |
+          valgrind build/bin/server_test
+
+      - name: Prepare, and Build Ωedit Integration Tests 🔧
+        run: |
+          cmake -S src/tests/integration -B build-tests-integration -DCMAKE_BUILD_TYPE=Debug -DCMAKE_PREFIX_PATH="${PWD}/_install"
+          cmake --build build-tests-integration
+
+      - name: Run Ωedit Integration Tests 📋
+        working-directory: build-tests-integration
+        run: |
+          ctest -C Debug --output-on-failure
+
+      - name: Build Tests 🔧
+        run: |
+          cmake -S src/tests -B build-tests -DCMAKE_BUILD_TYPE=Debug
+
+      - name: Run Tests 📋
+        if: runner.os != 'Windows'
+        working-directory: build-tests
+        run: |
+          ctest -C Debug --output-on-failure
+
+      - name: Upload Native JARs - Macos 🔺
+        uses: actions/upload-artifact@v3
+        if: runner.os == 'macOS'
+        with:
+          name: libomega_edit.dylib
+          path: lib/libomega_edit.dylib
+
+      - name: Upload Native JARs - Linux 🔺
+        uses: actions/upload-artifact@v3
+        if: runner.os == 'Linux'
+        with:
+          name: libomega_edit.so
+          path: lib/libomega_edit.so
+
+      - name: Upload Native JARs - Windows 🔺
+        uses: actions/upload-artifact@v3
+        if: runner.os == 'Windows'
+        with:
+          name: omega_edit.dll
+          path: lib/omega_edit.dll
+
+  scala-tests:
+    needs: [build-lib-cpp-tests]
+    strategy:
+      matrix:
+        os: [ macos-latest, ubuntu-latest ] # currently don't test windows as it seems to hang
```

Comment:
Windows hanging on the Scala API tests isn't a good sign.  We know it works under Windows because people are using it.  Wonder what's causing the hang.

Reply:
Will work this more in another different PR. Possibly after getting the TS client tests running